packages/autocertifier-client/src/errors.ts

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable max-len */
+
+import { ApiError } from './data/ApiError'
+import { HttpStatus } from './data/HttpStatus'
+
+export enum ErrorCode {
+    FAILED_TO_EXTRACT_IP_ADDRESS = 'FAILED_TO_EXTRACT_IP_ADDRESS',
+    UNSPECIFIED_ERROR = 'UNSPECIFIED_ERROR',
+    TOKEN_MISSING = 'TOKEN_MISSING',
+    STREAMR_WEBSOCKET_PORT_MISSING = 'STREAMR_WEBSOCKET_PORT_MISSING',
+    DATABASE_ERROR = 'DATABASE_ERROR',
+    INVALID_SUBDOMAIN_OR_TOKEN = 'INVALID_SUBDOMAIN_OR_TOKEN',
+    FAILED_TO_CONNECT_TO_STREAMR_WEBSOCKET = 'FAILED_TO_CONNECT_TO_STREAMR_WEBSOCKET',
+    SERVER_ERROR = 'SERVER_ERROR'
+}
+
+export class Err extends Error {
+
+    public code: ErrorCode
+    public httpStatus: HttpStatus
+    public originalError?: Error | string
+
+    constructor(code: ErrorCode, httpStatus: HttpStatus, message?: string, originalError?: Error | string) {
```

Comment:
Do we need to both `code` and `httpStatus`? The error code is a higher level error code which implies the underlying http status code?